Structure_Engine/Compute/MapTaperedProfile.cs

```diff
@@ -0,0 +1,267 @@
+                    IProfile newProfile;
+                    double newPosition = 0;
+
+                    if (position == 0)
+                    {
+                        //If the position is the same as the start of the TaperedProfile
+                        taperedProfile.Profiles.TryGetValue(0, out newProfile);
+                        newPosition = 0;
+                    }
+                    else if (position == 1)
+                    {
+                        //If the position is the same as the end of the TaperedProfile
+                        taperedProfile.Profiles.TryGetValue(1, out newProfile);
+                        newPosition = (1 - startPosition) / newLength;
+                    }
+                    else if (taperedProfile.Profiles.ContainsKey(position))
+                    {
+                        //If the position is equal to an existing position in the TaperedProfile
+                        taperedProfile.Profiles.TryGetValue(position, out newProfile);
+                        newPosition = (positions[i] - startPosition) / newLength;
+                    }
+                    else
+                    {
+                        //If the position is between existing positions in the TaperedProfile
+                        IProfile preProfile;
+                        IProfile postProfile;
+
+                        double prePosition = positions[i - 1];
+                        taperedProfile.Profiles.TryGetValue(prePosition, out preProfile);
+                        //If the both new positions are between original positions 
+                        if (preProfile == null)
+                        {
+                            taperedProfile.Profiles.TryGetValue(positions[i - 2], out preProfile);
+                            prePosition = positions[i - 2];
+                        }
+
+                        double postPosition = positions[i + 1];
+                        taperedProfile.Profiles.TryGetValue(postPosition, out postProfile);
+                        //If the both new positions are between original positions 
+                        if (postProfile == null)
+                        {
+                            taperedProfile.Profiles.TryGetValue(positions[i + 2], out postProfile);
+                            postPosition = positions[i + 2];
+                        }
+
+                        double interpolationPosition = (position - prePosition) / (postPosition - prePosition);
+
+                        int newInterpolationOrder;
+                        try
+                        {
+                            newInterpolationOrder = newInterpolationOrders[i];
+                        }
+                        catch (ArgumentOutOfRangeException)
+                        {
+                            //If it is the final index
+                            newInterpolationOrder = newInterpolationOrders[i - 1];
+                        }
```

Comment:
We should not use try-catch as a filter mechanism. Exceptions should only be used for real exceptions. They are significantly slower than doing the check needed ourselfs. Reading the comment of the code, I think something along the lines the following could work:


```suggestion
                        int interpolationIndex = Math.Min(i, newInterpolationOrders.Count-1);
                        int newInterpolationOrder = newInterpolationOrders[interpolationIndex];

```

Good to add some comment as well as to how the `i` relates to the list of interpolation orders. Assume you have one less interpolation order than number of profiles?